lib/widgets/content.dart

```diff
@@ -1117,23 +1124,26 @@ class _InlineContentBuilder {
 
 const kInlineCodeFontSizeFactor = 0.825;
 
-class UserMention extends StatelessWidget {
-  const UserMention({
+class Mention extends StatelessWidget {
+  const Mention({
     super.key,
     required this.ambientTextStyle,
     required this.node,
   });
 
   final TextStyle ambientTextStyle;
-  final UserMentionNode node;
+  final MentionNode node;
 
   @override
   Widget build(BuildContext context) {
     final contentTheme = ContentTheme.of(context);
     return Container(
       decoration: BoxDecoration(
         // TODO(#646) different for wildcard mentions
-        color: contentTheme.colorDirectMentionBackground,
+        color: switch (node) {
+          UserMentionNode() => contentTheme.colorDirectMentionBackground,
+          TopicMentionNode() => contentTheme.colorTopicMentionBackground,
```

Comment:
It seems like this is beginning to implement #646 — we're now giving some wildcard mentions (namely for topics) a different background vs. direct mentions.

Should @-channel mentions get the same background as @-topic?

Reply:
Yeah! But personal mentions trump channel and group mentions if a message has both.

Original issue for reference (we've probably tweaked colors since then): https://github.com/zulip/zulip/issues/22059
Corner case that needed fixing, in case this helps avoid an issue: https://github.com/zulip/zulip/issues/27654

Reply:
Thanks for the references!

To possibly clarify, though: the immediate code here is about how the mention is formatted within the message, rather than how the message as a whole gets highlighted when it is/has a mention. The latter is tracked under #647. I think that part will probably call for a separate PR from this one, but the formatting of the mention elements themselves might be easy to fold in here.

Reply:
Updated to remove this part, will open separate PR for #647 later as it's a post-launch issue.